ff/handlers/nonbonded.py

```diff
@@ -161,6 +162,82 @@ def parameterize_ligand(params, param_idxs):
     return params[param_idxs]
 
 
+def compute_or_load_am1_charges(mol):
+    """Unless already cached in mol's "AM1Cache" property, use OpenEye to compute AM1 partial charges."""
+
+    # check for cache
+    cache_key = "AM1Cache"
+    if not mol.HasProp(cache_key):
+        # The charges returned by OEQuacPac is not deterministic across OS platforms. It is known
+        # to be an issue that the atom ordering modifies the return values as well. A follow up
+        # with OpenEye is in order
+        # https://github.com/openforcefield/openff-toolkit/issues/983
+        am1_charges = list(oe_assign_charges(mol, "AM1"))
+
+        mol.SetProp(cache_key, base64.b64encode(pickle.dumps(am1_charges)))
+
+    else:
+        am1_charges = pickle.loads(base64.b64decode(mol.GetProp(cache_key)))
+
+    return np.array(am1_charges)
+
+
+def bond_smirks_matches(mol, smirks_list):
+    """Return an array of ordered bonds and an array of their assigned types
+
+    Notes
+    -----
+    * Uses OpenEye for substructure searches
+    * "First match wins" -- e.g. if bond (a,b) can be matched by smirks_list[2], smirks_list[5], ..., only return 2
+    """
+    oemol = convert_to_oe(mol)
+    AromaticityModel.assign(oemol)
+
+    bond_idxs = []  # [B, 2]
+    type_idxs = []  # [B]
+
+    for type_idx, smirks in enumerate(smirks_list):
+        matched_bonds = []
+
+        matches = oe_match_smirks(smirks, oemol)
+
+        for matched_indices in matches:
+            a, b = matched_indices[0], matched_indices[1]
+            forward_matched_bond = [a, b]
+            reverse_matched_bond = [b, a]
+
+            already_matched = forward_matched_bond in matched_bonds or reverse_matched_bond in matched_bonds
+            already_assigned = forward_matched_bond in bond_idxs or reverse_matched_bond in bond_idxs
+
+            if not (already_matched or already_assigned):
```

Comment:
Unless I'm missing something, `already_matched` implies `already_assigned`, so it seems like the former is not needed (nor is `matched_bonds`)?

Reply:
Hmm, I think you're right -- will confirm!

Initially just wanted to group and name the conditions checked in https://github.com/proteneer/timemachine/blob/d7acd339a681618fd2819b072f48f5aaafcf3b1b/ff/handlers/nonbonded.py#L392-L397

Reply:
Applied in https://github.com/proteneer/timemachine/pull/571/commits/4de08487903e12e9eaed524bc2a986895714ebda